### PR TITLE
Add `storedByte` to SiSPI driver's state

### PR DESCRIPTION
**Pull request by martijnbastiaan**
This essentially adds a register, making the component more likely to reach its timing requirements.

### CODE REVIEW

File: bittide/src/Bittide/ClockControl/Si539xSpi.hs [hunk 1]
```diff
@@ -225,6 +227,8 @@ data DriverState dom = DriverState
   -- ^ Current communication transaction.
   , commandAcknowledged :: Acknowledge
   -- ^ Whether or not the current transaction has already been acknowledged.
+  , storedBytes :: Maybe Byte
+  -- ^ Data we have received from the SPI interface.
```
Comment (hiddemoll) [inline bittide/src/Bittide/ClockControl/Si539xSpi.hs:231]: The name `storedBytes` (plural) does not match the type `Byte` (singular). But the commit message calls it `storedWord`. So what type should it be?

File: bittide/src/Bittide/ClockControl/Si539xSpi.hs [hunk 2]
```diff
@@ -301,12 +306,13 @@ si539xSpiDriver SNat incomingOpS miso = (fromSlave, decoderBusy, spiOut)
       | otherwise = satPred SatZero idleCycles
 
     nextState
-      | commandAcknowledged && not spiBusy =
-        DriverState nextPage nextAddress nextOp False maxBound
+      | commandAcknowledged && not spiBusy && isNothing storedBytes=
+        DriverState nextPage nextAddress nextOp False (fmap resize outBytes) maxBound
```
Comment (hiddemoll) [inline bittide/src/Bittide/ClockControl/Si539xSpi.hs:310]:
```suggestion
      | commandAcknowledged && not spiBusy && isNothing storedBytes =
        DriverState nextPage nextAddress nextOp False (fmap resize outBytes) maxBound
```

File: bittide/src/Bittide/ClockControl/Si539xSpi.hs [hunk 3]
```diff
@@ -301,12 +306,13 @@ si539xSpiDriver SNat incomingOpS miso = (fromSlave, decoderBusy, spiOut)
...
       | otherwise                          =
         currentState
           { commandAcknowledged = spiAck || commandAcknowledged
-          , idleCycles = updateIdleCycles}
+          , idleCycles = updateIdleCycles
+          , storedBytes = fmap resize outBytes}
```
Comment (hiddemoll) [inline bittide/src/Bittide/ClockControl/Si539xSpi.hs:315]: Here it is `outBytes` (plural) again

**Review by hiddemoll — CHANGES_REQUESTED** [review]

File: bittide/src/Bittide/ClockControl/Si539xSpi.hs (hunk 1 above)
Comment (martijnbastiaan) [reply bittide/src/Bittide/ClockControl/Si539xSpi.hs:231]: Just discussed this in the office: `storedByte`.

File: bittide/src/Bittide/ClockControl/Si539xSpi.hs (hunk 2 above)
Comment (martijnbastiaan) [reply bittide/src/Bittide/ClockControl/Si539xSpi.hs:310]: Turns out this is actually multiple bytes (`BitVector 16`). The resize just takes the byte at the LSBs.

**Review by hiddemoll — APPROVED** [review]

**Pull request merged**